### DISCUSSION TITLE
API Authorisation in Python

### DISCUSSION THREAD

**Discussion by edwapj** [amberelectric/public-api#23] [Q&A]
I use the requests and json libraries in Python to access the old Amber API.

payload = '{ "postcode": "5000"}'                         # post code no longer required for the new public API
headers = {'Content-Type': 'application/json'}
response = requests.request("POST", url, headers=headers, data = payload)

How can I include authorisation with my token for the new Amber API in the requests header ?

**Comment by madpilot** [comment]
```python
api_token = 'psk_...'
headers={'Content-Type': 'application/json', 'Authorization': 'Bearer ' + api_token}
response = requests.request("GET", url, headers=headers)
```

Though I have a beta python library if that helps: https://github.com/madpilot/amberelectric.py

I haven't turned it into a pip yet - I want to soak test it a little bit first, but it should be complete. The docs probably aren't :)

**Reply by edwapj** [reply]
Thanks for your answer
I can now access the new Amber API in Python. 
I have learnt that the trailing space in 'Bearer ' is essential. 
Also 'Authorization' is not spelt with a 's' in Python although it does not raise a syntax error.

I will be interested to try our your beta python library

**Reply by madpilot** [reply]
Yeah, that is part of the HTTP spec. Thanks America.
